### ISSUE TITLE
[Feature Request] Save checkpoint of best performing model

### ISSUE THREAD

**Issue by StevenShi-23** [KaiyangZhou/deep-person-reid#284]
Hi Kaiyang, 

Thanks for the amazing work! This project is very easy to follow and well-documented, which makes it easier to develop our own models.

I wonder if we support saving best performing model (testing or training) instead of every intermediate models. In current implementation, it seems that every evaluated model will be dumped into checkpoint, which makes the log directory very storage consuming. I've done a brief scan but did not find a way/issue to address it. 

This can be achieve by simply changing `save_checkpoint` under `utils/torchtools.py`, and adding a flag like `save_best_model: ["train", "test"]` in top level config. If you have plan to add this feature in the future, I am happy to help as well. 

Thanks.

**Comment by KaiyangZhou** [comment]
sounds good, you are welcome to submit a PR :)

an easy way to keep track of the best performing model is to add `is_best` in this line https://github.com/KaiyangZhou/deep-person-reid/blob/master/torchreid/engine/engine.py#L145 but this still saves the intermediate model

**Comment by StevenShi-23** [comment]
Cool, I'm working on it.

Besides, I have another issue with train set's relabelling. Sometimes it will be helpful if we can have new pid dumped into a file after relabelling, because this mapping is not known to users once training is finished. By saving the mapping <pid_old, pid_new>, we can make use of predictions on training set. 

An example implementation is [here](https://github.com/KaiyangZhou/deep-person-reid/compare/master...StevenShi-23:feat_save_relabel). Please let me know how you think. Thanks!
